tests/reference/cpp-expr10-76fb526.stdout

```diff
@@ -0,0 +1,30 @@
+#include <iostream>
+#include <string>
+#include <vector>
+#include <Kokkos_Core.hpp>
+#include <lfortran_intrinsics.h>
+
+template <typename T>
+Kokkos::View<T*> from_std_vector(const std::vector<T> &v)
+{
+    Kokkos::View<T*> r("r", v.size());
+    for (size_t i=0; i < v.size(); i++) {
+        r(i) = v[i];
+    }
+    return r;
+}
+
+void test_UnaryOp()
+{
+    int a;
+    bool b1;
+    bool b2;
+    float f;
+    a = 4;
+    a = -500;
+    f = 1.000000;
+    f = -183745.534000;
+    b1 = true;
+    b2 = !false;
```

Comment:
I'm gonna explore more of the c++ codegen part for some time, from now on. This can help me give a break from Python AST->ASR. :)
Let me know if this looks good.

Reply:
I am unsure, as it seems the line `result = !op_value;` is not evaluated in this case. It seems it is actually represented as UnaryOp, as it should be. But your code in this PR seems to instead return the actual answer so not `!false`, but `true`.

Reply:
> I am unsure, as it seems the line `result = !op_value;` is not evaluated in this case. It seems it is actually represented as UnaryOp, as it should be. But your code in this PR seems to instead return the actual answer so not `!false`, but `true`.

That line, `result = !op_value;`, is being evaluated since here is the node that was created:
```
               (=
                  (Var 2 b2)
                  (UnaryOp
                     Not
                     (ConstantLogical .false.
                        (Logical 1 []))
                     (Logical 1 [])
                     (ConstantLogical .true.
                        (Logical 1 []))) ())]
```
With `.true.` passed in as the value member. But I'm not sure what we want the cpp code to produce: `true` or `!false`.

Reply:
Am I missing something?

Reply:
Ah, so it is already doing what it should! Ok, that's fine. The C++ code should return `!false`. The backend returns the full expression by default.